src/parser/parser-office.ee.ts

```diff
@@ -0,0 +1,146 @@
+import { XMLParser } from 'fast-xml-parser';
+
+/**
+ * Parse a document file and extract text content
+ * @param documents All required documents
+ * @returns Extracted text as a string
+ */
+export async function extractOfficeDocument(documents: {
+  [key: string]: Uint8Array;
+}): Promise<string> {
+  const parser = new XMLParser({
+    alwaysCreateTextNode: true,
+    ignoreAttributes: true,
+    preserveOrder: true,
+    removeNSPrefix: true,
+  });
+
+  let combinedText = '';
+
+  for (const [, content] of Object.entries(documents)) {
+    const parsed = parser.parse(
+      new TextDecoder('utf8').decode(content),
+    ) as XMLNode;
+
+    const extractedText = await recursiveXMLParsedExtract({
+      source: parsed,
+      text: '',
+    });
+
+    combinedText += `${formatText(extractedText)}\n`;
+  }
+
+  return combinedText.trim();
+}
+
+/**
+ * This function takes an unknown source, and will recursively try to extract text from it.
+ * Handles paragraphs and headers separately to preserve newlines.
+ * @param root named params
+ * @param root.source an object or a list to extract text from
+ * @param root.text a string to concatenate with the result of this function. Set to undefined to create a new string
+ * @returns either undefined or a string concatenated with root.text
+ */
+async function recursiveXMLParsedExtract({
+  source,
+  text,
+}: {
+  source: XMLNode;
+  text: string | undefined;
+}): Promise<string | undefined> {
```

Comment:
Your final code for the function definition (look at my other comments)

**Note** the function is no longer async, because there is no need for it

```ts
function recursiveXMLParsedExtract({
  source,
  text,
}: {
  source: XMLNode;
  extractedText: string;
}): void {
  // ...
}
```